### ISSUE TITLE
Support for OSX

### ISSUE THREAD

**Issue by robotframework-ride** [robotframework/RIDE#67]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33) by @pekkaklarck on 13 Oct 2008*

The current version of RIDE doesn't even start on OSX because some of the
wx components it uses aren't supported on the OSX version of wxPython. We
need to investigate what components are missing and can we replace them
with others.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c1) by valkolovos on 27 Apr 2009*


0.13.1 starts but fails to load the test editor. WxPython on OS X apparently does not
support PopupWindows.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c2) by @pekkaklarck on 27 Apr 2009*


We've been thinking about switching from wxPython to PyQT. Does anyone here know how
well PyQT4 works on OS X? General comments about pros/cons for PyQT are also
appreciated here or on mailing lists.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c3) by bryan.oakley on 28 Apr 2009*


Why do you not consider using Tkinter? Modern versions of Tkinter have everything you
would likely need for Ride. I know Tk has a bad reputation but I think that
reputation is undeserved. If you're not considering it for some specific reason would
you be willing to share why?

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c4) by @pekkaklarck on 28 May 2009*


It should be possible to use RIDE on OSX after <a href="https://code.google.com/p/robotframework-ride/source/detail?r=397">r397</a>. Keyword completion and
documentation popups don't work, though, but I opened a separate <a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=146" title="Keyword completion and documentation popups don't work on OSX at all"> issue 146 </a> about it.
We still need to test the change on other platforms before we can close this issue.
If everything works well, we'll probably create 0.15.2 version with this fix soonish.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c6) by @pekkaklarck on 28 May 2009*


After brief testing it seems that most of the features work on OSX relatively well.
Problems that we faced are:

1) The dialog used for editing setup/teardown doesn't work. This is probably due to
this dialog using keyword completion.

2) Navigating to user keywords using ctrl doesn't work. Double clicking a keyword
name works, though, and ctrl is able to underline the keyword name.

3) Key-bindings are inconsistent -- some use ctrl and some cmd key.

4) As know and noted before, keyword completion and documentation pop-ups don't work
(<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=146" title="Keyword completion and documentation popups don't work on OSX at all"> issue 146 </a>). Keyword completion is able to select the first matching keyword so is
partly usable.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c7) by @pekkaklarck on 28 May 2009*


We won't have time to investigate this further this week. Any help and further
investigation is highly appreciated!

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c8) by @yanne on 28 May 2009*


After !) has been resolved, we should make a release, because that version would
allow editing all data in OSX

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c9) by @pekkaklarck on 3 Jun 2009*


We won't do 0.15.2 but go straight to 0.16.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c10) by @yanne on 12 Jun 2009*


This issue can be considered done, data can be now opened and edited also in OSX.

**Comment by robotframework-ride** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=33#c11) by @pekkaklarck on 12 Jun 2009*


There are some missing features (<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=146" title="Keyword completion and documentation popups don't work on OSX at all"> issue 146 </a>) and probably also uncovered bugs on OS X,
but they can be tracked separately. Great to have basic OSX support now!
